--- a/XADMacArchiveParser.m
+++ b/XADMacArchiveParser.m
@@ -405,6 +405,8 @@
 	[template setObject:[NSNumber numberWithUnsignedInt:CSUInt32BE(bytes+65)] forKey:XADFileTypeKey];
 	[template setObject:[NSNumber numberWithUnsignedInt:CSUInt32BE(bytes+69)] forKey:XADFileCreatorKey];
 	[template setObject:[NSNumber numberWithInt:bytes[101]+(bytes[73]<<8)] forKey:XADFinderFlagsKey];
+	[template setObject:[NSNumber numberWithUnsignedInt:CSUInt32BE(bytes+91)] forKey:@"PacksmithCreated1904"];
+	[template setObject:[NSNumber numberWithUnsignedInt:CSUInt32BE(bytes+95)] forKey:@"PacksmithModified1904"];
 	[template setObject:[NSDate XADDateWithTimeIntervalSince1904:CSUInt32BE(bytes+91)] forKey:XADCreationDateKey];
 	[template setObject:[NSDate XADDateWithTimeIntervalSince1904:CSUInt32BE(bytes+95)] forKey:XADLastModificationDateKey];
 	[template removeObjectForKey:XADDataLengthKey];
